libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/IpvSessionService.java

```diff
@@ -51,52 +52,54 @@ public IpvSessionService(ConfigService configService) {
     }
 
     public IpvSessionItem getIpvSession(String ipvSessionId) {
-        return dataStore.getItem(ipvSessionId);
+        return getIpvSession(ipvSessionId, false);
     }
 
-    public Optional<IpvSessionItem> getIpvSessionByAuthorizationCode(String authorizationCode) {
-        IpvSessionItem ipvSessionItem =
-                dataStore.getItemByIndex(
-                        "authorizationCode", DigestUtils.sha256Hex(authorizationCode));
-        return Optional.ofNullable(ipvSessionItem);
-    }
-
-    public IpvSessionItem getIpvSessionByAccessToken(String accessToken)
-            throws UnknownAccessTokenException, GetAccessTokenException {
-        try {
-            return Retry.runTaskWithBackoff(
-                    sleeper,
-                    7,
-                    10,
+    public IpvSessionItem getIpvSession(String ipvSessionId, boolean withRetry)
```

Comment:
Is this ever called with `withRetry = false`? Should we just rename it `getIpvSessionWithRetry`?

Reply:
called frm original method - getIpvSession(String ipvSessionId)

Reply:
Yeah, but that's the only place, so instead you could do:
```
public IpvSessionItem getIpvSession(String ipvSessionId) {
    return dataStore.getItem(ipvSessionId);
}

public IpvSessionItem getIpvSessionWithRetry(String ipvSessionId)
        throws IpvSessionNotFoundException {
    return callRunTaskWithBackoff(...);
}
```
You still only have one copy of each implementation, and the non-retried version doesn't need to declare an exception that it'll never throw...

Reply:
Done